pkg/build/api/v1/types.go

```diff
@@ -52,8 +52,73 @@ type BuildSpec struct {
 	// scheduled in the system, that the build may be active on a node before the
 	// system actively tries to terminate the build; value must be positive integer
 	CompletionDeadlineSeconds *int64 `json:"completionDeadlineSeconds,omitempty"`
+
+	// Info is a placeholder that can store multiple reasons why a build was
+	// triggered. It should cover for a future implementation of level triggered
+	// builds where you can multiple reasons why a build was triggered.
+	Info []*BuildTriggerInfo `json:"info,omitempty"`
```

Comment:
I'm not really happy with the name `Info` here.  Are you in a state where you are ready for API review?